### PR TITLE
chapter 7

### PR DESCRIPTION
**Pull request by truongha-1240**
Chapter 7: Sign Up
☑︎ Create Showing user
☑︎ Create Signup form
☑︎ Unsuccessful signups
☑︎ Successful signups
☑︎ Add i18n for error messages
image

### CODE REVIEW

**Comment by truongha-1240** [comment]
ready

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  def index; end
+
+  def show
+    @user = User.find(params[:id])
+    if @user
+      @user
+    else
+      flash[:danger] = t "noti.error.user_not_found"
+      redirect_to root_path
+    end
```
Comment (vanvtt-0952) [inline app/controllers/users_controller.rb]:
```suggestion
    @user = User.find_by id: params[:id]
    return if @user
    
    flash[:danger] = t "noti.error.user_not_found"
    redirect_to root_path
```

**Comment by truongha-1240** [comment]
ready

**Comment by truongha-1240** [comment]
merged

**Pull request merged**